jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetSamDebugSupport.kt

```diff
@@ -0,0 +1,305 @@
+
+        commandLine.withParameters("--debugger-path")
+                .withParameters(binDir.path)
+                .withParameters("--debug-args")
+                .withParameters(debugArgs)
+
+        super.patchCommandLine(debugPort, state, commandLine)
+    }
+
+    override fun createDebugProcess(
+        environment: ExecutionEnvironment,
+        state: SamRunningState,
+        debugPort: Int
+    ): XDebugProcessStarter? {
+        throw UnsupportedOperationException("Use 'createDebugProcessAsync' instead")
+    }
+
+    override fun createDebugProcessAsync(
+        environment: ExecutionEnvironment,
+        state: SamRunningState,
+        debugPort: Int
+    ): Promise<XDebugProcessStarter?> {
+
+        val promise = AsyncPromise<XDebugProcessStarter?>()
+
+        val project = environment.project
+        val debuggerLifetimeDefinition = project.defineNestedLifetime()
+        val debuggerLifetime = debuggerLifetimeDefinition.lifetime
+
+        val scheduler = RdDispatcher(debuggerLifetime)
+        val startInfo = createNetCoreStartInfo(state)
+
+        val protocol = Protocol(
+            serializers = Serializers(),
+            identity = Identities(IdKind.Client),
+            scheduler = scheduler,
+            wire = SocketWire.Client(debuggerLifetime, scheduler, port = debugPort, optId = "FrontendToDebugWorker"),
+            lifetime = debuggerLifetime
+        )
+
+        val executionResult = state.execute(environment.executor, environment.runner)
+
+        protocol.wire.connected.adviseUntil(debuggerLifetime) connected@{ isConnected ->
+            if (!isConnected) {
+                return@connected false
+            }
+
+            try {
+                val workerModel = RiderDebuggerWorkerModelManager.createDebuggerModel(debuggerLifetime, protocol)
+
+                workerModel.initialized.adviseUntil(debuggerLifetime) initialized@{ isInitialized ->
+                    if (!isInitialized) {
+                        return@initialized false
+                    }
+
+                    // Fire backend to connect to debugger.
+                    environment.project.solution.debuggerWorkerConnectionHelperModel.ports.put(
+                        debuggerLifetime,
+                        environment.executionId,
+                        debugPort
+                    )
+
+                    val sessionModel = DotNetDebuggerSessionModel(startInfo)
+                    sessionModel.sessionProperties.bindToSettings(debuggerLifetime).apply {
+                        enableHeuristicPathResolve.set(true)
+                    }
+
+                    workerModel.activeSession.set(sessionModel)
+                    val console =
+                        TextConsoleBuilderFactory.getInstance().createBuilder(environment.project).console
+                    val processHandler = object : ProcessHandler() {
+                        override fun detachProcessImpl() {
+                            val process =
```

Comment:
can this just call destroyProcessImpl() since logic is the same

Reply:
Fixed.